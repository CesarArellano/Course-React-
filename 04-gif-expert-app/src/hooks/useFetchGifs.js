import React, { useState } from 'react'

export const useFetchGifs = () => {
  const [state, setstate] = useState({
    data: [],
    loading: true
  });

  setTimeout(()=> {
    setstate({
      data: [1,2,2,323,2],
      loading: false
    });
  }, 3000);

  return state;
}
