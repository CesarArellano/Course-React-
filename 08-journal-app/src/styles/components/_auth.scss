.auth__main {
  align-items: center;
  background: $primary;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.auth__box-container {
  background: #fff;
  box-shadow: 0px 2px 5px $dark-grey;
  border-radius: 5px;
  padding: 20px;
  width: 250px;
}

.auth__title {
  color: $primary;
}