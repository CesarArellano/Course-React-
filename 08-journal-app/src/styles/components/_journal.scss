.journal__main-content {
  display: flex;
}

.journal__sidebar {
  background: $dark-grey;
  color: white;
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 450px;
  padding-left: 10px;
  padding-right: 10px;
}

.journal__sidebar-navbar {
  display: flex;
  justify-content: space-between;

  h3 {
    font-weight: lighter;
  }
}

.journal__new-entry {
  align-items: center;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 30px;
  width: 100%;
  transition: 0.4s ease-out;

  &:hover {
    color: darken($color: white, $amount: 20)
  }
  
}

.journal__entries {
  flex: 1 1 auto;
  padding: 5px;
  margin-top: 30px;
  overflow-y: scroll;
}

.journal__entry {
  padding: 5px;
  background: white;
  border-radius: 4px;
  color: $dark-grey;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
  overflow: hidden;
}

.journal__entry-picture {
  height: 100px;
  width: 100px;
  object-fit: cover;
}

.journal__entry-body {
  padding: 10px;
}

.journal__entry-title {
  font-size: 14px;
  font-weight: bold;
}

.journal__entry-content {
  font-size: 10px;
}

.journal__entry-date-box {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 5px;

  span {
    font-size: 12px;
  }
}
