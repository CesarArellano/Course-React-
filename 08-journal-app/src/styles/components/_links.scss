.link {
  color: $dark-grey;
  text-decoration: none;
  transition: 1s ease;
  
  &:hover {
    border-bottom: 1px solid gray;
  }
}