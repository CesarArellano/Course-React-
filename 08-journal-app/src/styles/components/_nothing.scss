.nothing__main-content {
  align-items: center;
  background: $primary;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100vh;
  

  p {
    text-align: center;
  }

  i {
    transition: 0.5s cubic-bezier(0.645, 0.045, 0.355, 1);
  }

  i:hover {
    color:yellow;
    transform: rotate(360deg);
  }
}