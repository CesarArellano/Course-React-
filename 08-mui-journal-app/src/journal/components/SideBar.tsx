import { TurnedInNot } from '@mui/icons-material'
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from '@mui/material'
import React from 'react'

interface Props {
  drawerWidth: number
}

export const SideBar = ({ drawerWidth }: Props) => {
  return (
    <Box
      component="nav"
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 }}}
    >
      <Drawer
        variant="permanent"
        open
        color='primary.main'
        sx={{
          display: { xs: 'none', sm: 'block' },
          '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">CÃ©sar Arellano</Typography>
        </Toolbar>
        <Divider />
        <List>
          {
            ['Enero', 'Febrero', 'Marzo', 'Abril'].map(text => (
              <ListItem key={ text } disablePadding>
                <ListItemButton>
                  <ListItemIcon>
                    <TurnedInNot />
                  </ListItemIcon>
                  <Grid container>
                    <ListItemText primary={ text } />
                    <ListItemText secondary={ 'Et ullamco id ut aliquip velit duis quis aliquip id nostrud cillum aliqua eu voluptate.' } />
                  </Grid>
                </ListItemButton>
              </ListItem>
            ))
          }
        </List>
      </Drawer>
    </Box>
  )
}
