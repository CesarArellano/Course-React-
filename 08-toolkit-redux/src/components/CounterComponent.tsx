import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { decrement, increment, reset } from '../store/slices/counter/counterSlice';

export const CounterComponent = () => {
  const { value } = useSelector<any, any>((state) => state.counter);
  const dispatch = useDispatch();
  
  const handleAdd = () => {
    dispatch( increment() );
  };

  const handleSubstract = () => {
    if(value > 0) {
      dispatch( decrement() );
    }    
  }

  const handleReset = () => {
    dispatch( reset() )
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{ value }</h2>
      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleSubstract }>-1</button>
      <button onClick={ handleReset }>Reset</button>
    </>
  );
}
