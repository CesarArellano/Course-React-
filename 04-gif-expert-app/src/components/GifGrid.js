import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data, loading} = useFetchGifs();
  console.log(loading);
  // useEffect(() => {
  //   getGifs( category )
  //     .then( setImages );
  // }, [ category ]);
  

  return (
    <>
      <h3>{ category }</h3>
      {loading ? 'Cargando' : 'Se carg√≥ exitosamente'}
      {/* <div className="card-grid">
        {
          images.map(( img ) => (
            <GifGridItem
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div> */}
    </>
  )
}
