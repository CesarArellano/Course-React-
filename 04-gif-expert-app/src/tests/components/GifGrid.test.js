import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid Component', () => {
  const category = 'One Punch';

  test('Must display correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true 
    });
    const wrapper = shallow(<GifGrid category = { category } />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Must display items when loading images of useFetchGifs', () => {
    const gifs = [
      {
        id: '223123',
        url: 'https://i.blogs.es/65e246/traductor-de-google/450_1000.jpg',
        title: 'Google'
      },
      {
        id: '29879843',
        url: 'https://i.blogs.es/65e246/traductor-de-apple/450_1000.jpg',
        title: 'Apple'
      },
  ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category = { category } />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
})
