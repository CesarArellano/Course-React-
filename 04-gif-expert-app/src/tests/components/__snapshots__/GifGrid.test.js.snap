// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid Component Must display correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate_flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid Component Must display items when loading images of useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="223123"
      key="223123"
      title="Google"
      url="https://i.blogs.es/65e246/traductor-de-google/450_1000.jpg"
    />
    <GifGridItem
      id="29879843"
      key="29879843"
      title="Apple"
      url="https://i.blogs.es/65e246/traductor-de-apple/450_1000.jpg"
    />
  </div>
</Fragment>
`;
